lib/src/front_end/comment_writer.dart

```diff
@@ -312,7 +313,7 @@ class CommentSequence extends ListBase<SourceComment> {
   SourceComment operator [](int index) => _comments[index];
 
   @override
-  operator []=(int index, SourceComment value) =>
+  void operator []=(int index, SourceComment value) =>
```

Comment:
This return type is pointless. Index setters are like setters: they can never return a value anyway.

Reply:
Generally agree, so maybe the lint needs to be updated for this case!

Reply:
I believe the lint is doing the right thing. I enabled always_declare_return_types and tested it on:

```dart
class Foo {
  get getter => 1;
  set setter(int value) => 1;
  method() => 1;
  operator [](int index) => 1;
  operator []=(int index, int value) => 1;
}
```

I get lint warnings on `getter`, `method()`, and `operator []`, but not `setter` or `operator []=`. Maybe the fix is out of sync with the lint? Or maybe this was just a mistake fixing the lint? Either way, I think you can delete this `void` and it should still be lint clean.